picasso/src/test/java/com/squareup/picasso/MarkableInputStreamTest.java

```diff
@@ -52,6 +52,35 @@ public void test() throws Exception {
     }
   }
 
+  @Test
+  public void exceedLimitTest() throws Exception {
+    MarkableInputStream in = new MarkableInputStream(new ByteArrayInputStream(
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ".getBytes(Charset.forName("US-ASCII"))),true);
+    assertThat(readBytes(in, 3)).isEqualTo("ABC");
+    long posA = in.savePosition(7);// DEFGHIJ
+    assertThat(readBytes(in, 4)).isEqualTo("DEFG");
+    in.mark(5); // HIJKL
+    assertThat(readBytes(in, 4)).isEqualTo("HIJK");
+    in.reset(); // Back to 'H'
+    assertThat(readBytes(in, 3)).isEqualTo("HIJ");
+    in.reset(posA); // Back to 'D'
+    assertThat(readBytes(in, 7)).isEqualTo("DEFGHIJ");
+    in.reset(); // Back to 'H' again.
+    assertThat(readBytes(in, 6)).isEqualTo("HIJKLM");
+    try {
+      in.reset(); // Back to 'H' again despite the limit being exceeded
+      assertThat(readBytes(in, 2)).isEqualTo("HI"); // confirm we're really back at H
+    } catch (IOException notexpected) {
```

Comment:
You can delete both of these try/catch blocks. If an exception is thrown it'll fail the test.